Library/GeneralImage.cpp

```diff
@@ -296,6 +299,76 @@ bool GeneralImage::LoadImage(const std::wstring& imageName)
 	return false;
 }
 
+bool GeneralImage::LoadImageFromMeasure(Measure* m)
+{
+	if (!m_Skin || m == nullptr)
+	{
+		DisposeImage();
+		return false;
+	}
+
+	auto pm = dynamic_cast<MeasurePlugin*>(m);
+	if (pm == nullptr)
+	{
+		DisposeImage();
+		return false;
+	}
+
+	ImageOptions info;
+	info.m_Path = std::wstring{ L"measure://" } + m->GetSkin()->GetFilePath() + L"/" + m->GetName();
+	info.m_FileSize = 0;
+
+	// PluginImageData is a type to be created, must contain at least:
+	// 1. uint32_t pixels array
+	// 2. Width and height
+	// 3. uint64_t timestamp

```

Comment:
What's the need for `timestamp`?

Reply:
Timestamp is part of image cache key.
The easiest way to add in-memory transfer would be to just reuse this cache, because as I see a lot of code depends on it. And if we want to use the cache, we should have a timestamp. We will say that plugin should update the timestamp whenever image is updated. Doesn't have to be an actual timestamp, simple `timestamp++` would be enough.